qiskit_ibm_runtime/noise_learner/noise_learner.py

```diff
@@ -0,0 +1,257 @@
+                the other fields (such as ``observables`` and ``parameter_values``) are ignored.
+
+        Returns:
+            The submitted job.
+
+        """
+        if not all(isinstance(t, QuantumCircuit) for t in circuits):
+            coerced_pubs = [EstimatorPub.coerce(pub) for pub in circuits]
+            circuits = [p.circuit for p in coerced_pubs]
+
+        # Store learner-specific and runtime options in different dictionaries
+        options_dict = asdict(self.options)
+        learner_options = {"options": self._get_inputs_options(options_dict)}
+        runtime_options = NoiseLearnerOptions._get_runtime_options(options_dict)
+
+        # Define the program inputs
+        inputs = {"circuits": circuits}
+        inputs.update(learner_options)
+
+        if self._backend:
+            for task in circuits:
+                if getattr(self._backend, "target", None) and not is_simulator(self._backend):
+                    validate_isa_circuits([task], self._backend.target)
+
+                if isinstance(self._backend, IBMBackend):
+                    self._backend.check_faulty(task)
+
+        logger.info("Submitting job using options %s", learner_options)
+
+        # Batch or Session
+        if self._mode:
+            return self._mode.run(
+                program_id=self._program_id(),
+                inputs=inputs,
+                options=runtime_options,
+                callback=options_dict.get("environment", {}).get("callback", None),
+                result_decoder=DEFAULT_DECODERS.get(self._program_id()),
+            )
+
+        if self._backend:
+            runtime_options["backend"] = self._backend
+            if "instance" not in runtime_options and isinstance(self._backend, IBMBackend):
+                runtime_options["instance"] = self._backend._instance
+
+        if isinstance(self._service, QiskitRuntimeService):
+            return self._service.run(
+                program_id=self._program_id(),
+                options=runtime_options,
+                inputs=inputs,
+                callback=options_dict.get("environment", {}).get("callback", None),
+                result_decoder=DEFAULT_DECODERS.get(self._program_id()),
+            )
+
+        return self._service.run(  # type: ignore[attr-defined]
+            program_id=self._program_id(),  # type: ignore[arg-type]
+            options=runtime_options,
+            inputs=inputs,
+        )
+
+    @classmethod
+    def _program_id(cls) -> str:
+        """Return the program ID."""
+        return "noise-learner"
+
+    def _set_options(
+        self, options: Optional[Union[Dict, NoiseLearnerOptions, EstimatorOptions]] = None
+    ) -> None:
+        """
+        Sets the options, ensuring that they are of type ``NoiseLearnerOptions``.
+        """
+        if not options:
+            self._options = NoiseLearnerOptions()
+        elif isinstance(options, NoiseLearnerOptions):
+            self._options = options
```

Comment:
I see `self._options = replace(options)` in BasePrimitiveV2, so this non-copy behaviour may be unexpected.

Reply:
okay, i can replace!